File: pkg/expo/doctor.go
package expo

import (
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"
	"strconv"
	"strings"

	"github.com/clix-so/clix-cli/pkg/logx"
)

// RunDoctor performs comprehensive checks for React Native Expo Clix SDK setup
func RunDoctor() error {
	projectRoot, err := os.Getwd()
	if err != nil {
		return fmt.Errorf("failed to get current working directory: %v", err)
	}

	logx.Log().Title().Println("Running Clix Doctor for React Native Expo…")
	logx.Separatorln()

	var issues []string

	// Check 1: Expo project structure
	if !CheckExpoProject(projectRoot) {
		issues = append(issues, "Not an Expo project - missing app.json or expo dependency")
	} else {
		logx.Log().Success().Println("Expo project detected")
	}

	// Check 2: Required dependencies
	missingDeps := CheckDependencies(projectRoot)
	if len(missingDeps) > 0 {
		issues = append(issues, fmt.Sprintf("Missing dependencies: %s", strings.Join(missingDeps, ", ")))
	} else {
		logx.Log().Success().Println("All required dependencies installed")
	}

	// Check 3: Firebase configuration files
	hasAndroidConfig := CheckFirebaseConfig(projectRoot, "android")
	hasIOSConfig := CheckFirebaseConfig(projectRoot, "ios")

	if !hasAndroidConfig {
		issues = append(issues, "Missing google-services.json file")
	} else {
		logx.Log().Success().Println("google-services.json found")
	}

	if !hasIOSConfig {
		issues = append(issues, "Missing GoogleService-Info.plist file")
	} else {
		logx.Log().Success().Println("GoogleService-Info.plist found")
	}

	// Check 4: app.json configuration
	configIssues := CheckAppConfig(projectRoot)
	if len(configIssues) > 0 {
		issues = append(issues, configIssues...)
	} else {
		logx.Log().Success().Println("app.json properly configured")
	}

	// Check 5: Clix initialization file
	if !CheckClixInitialization(projectRoot) {
		issues = append(issues, "Clix initialization file not found")
	} else {
		logx.Log().Success().Println("Clix initialization file found")
	}

	// Check 6: Clix integration in App component
	if !CheckClixAppIntegration(projectRoot) {
		issues = append(issues, "Clix not integrated in App component")
	} else {
		logx.Log().Success().Println("Clix integrated in App component")
	}

	// Check 6: Generated native code
	if !CheckNativeCode(projectRoot) {
		issues = append(issues, "Native code not generated - run 'npx expo prebuild --clean'")
	} else {
		logx.Log().Success().Println("Native code generated")
	}

	// Report results
	logx.NewLine()
	if len(issues) == 0 {
		logx.Log().Success().Println("All checks passed! Your Expo project is ready for Clix SDK.")
		logx.Log().Info().Println("You can now run 'npx expo run:android' or 'npx expo run:ios' to test push notifications.")
	} else {
		logx.Log().Failure().Println(fmt.Sprintf("Found %d issue(s):", len(issues)))
		for i, issue := range issues {
			logx.Log().Indent(2).Println(fmt.Sprintf("%d. %s", i+1, issue))
		}
		logx.NewLine()
		logx.Log().Warn().Println("Please fix the above issues and run 'clix doctor --expo' again.")
	}

	return nil
}

// CheckDependencies checks if all required dependencies are installed
func CheckDependencies(projectRoot string) []string {
	requiredDeps := []string{
		"expo-dev-client",
		"@react-native-firebase/app",
		"@react-native-firebase/messaging",
		"expo-build-properties",
		"@clix-so/react-native-sdk",
		"@notifee/react-native",
		"react-native-device-info",
		"react-native-get-random-values",
		"uuid",
	}

	packageJSONPath := filepath.Join(projectRoot, "package.json")
	data, err := os.ReadFile(packageJSONPath)
	if err != nil {
		return requiredDeps // Return all as missing if can't read package.json
	}

	var packageJSON map[string]any
	if err := json.Unmarshal(data, &packageJSON); err != nil {
		return requiredDeps
	}

	dependencies := make(map[string]bool)
	if deps, ok := packageJSON["dependencies"].(map[string]any); ok {
		for dep := range deps {
			dependencies[dep] = true
		}
	}
	if devDeps, ok := packageJSON["devDependencies"].(map[string]any); ok {
		for dep := range devDeps {
			dependencies[dep] = true
		}
	}

	var missing []string
	for _, dep := range requiredDeps {
		if !dependencies[dep] {
			missing = append(missing, dep)
		}
	}

	// Check MMKV separately as it needs version-specific validation
	if !checkMMKVVersion(projectRoot, dependencies) {
		missing = append(missing, "react-native-mmkv (incorrect version)")
	}

	return missing
}

// CheckAppConfig checks if app.json is properly configured
func CheckAppConfig(projectRoot string) []string {
	var issues []string

	appJSONPath := filepath.Join(projectRoot, "app.json")
	data, err := os.ReadFile(appJSONPath)
	if err != nil {
		issues = append(issues, "Could not read app.json")
		return issues
	}

	var config AppConfig
	if err := json.Unmarshal(data, &config); err != nil {
		issues = append(issues, "Could not parse app.json")
		return issues
	}

	// Check for Firebase App plugin
	hasFirebaseAppPlugin := false
	hasFirebaseMessagingPlugin := false
	for _, plugin := range config.Expo.Plugins {
		if pluginStr, ok := plugin.(string); ok {
			if pluginStr == "@react-native-firebase/app" {
				hasFirebaseAppPlugin = true
			}
			if pluginStr == "@react-native-firebase/messaging" {
				hasFirebaseMessagingPlugin = true
			}
		}
		if pluginArray, ok := plugin.([]any); ok && len(pluginArray) > 0 {
			if pluginStr, ok := pluginArray[0].(string); ok {
				if pluginStr == "@react-native-firebase/app" {
					hasFirebaseAppPlugin = true
				}
				if pluginStr == "@react-native-firebase/messaging" {
					hasFirebaseMessagingPlugin = true
				}
			}
		}
	}

	if !hasFirebaseAppPlugin {
		issues = append(issues, "@react-native-firebase/app plugin not configured in app.json")
	}

	if !hasFirebaseMessagingPlugin {
		issues = append(issues, "@react-native-firebase/messaging plugin not configured in app.json")
	}

	// Check for expo-build-properties plugin and its configuration
	hasBuildPropertiesPlugin := false
	hasIOSUseFrameworks := false
	hasAndroidExtraMavenRepos := false

	for _, plugin := range config.Expo.Plugins {
		if pluginStr, ok := plugin.(string); ok && pluginStr == "expo-build-properties" {
			hasBuildPropertiesPlugin = true
			// String plugin format doesn't have configuration, so these are missing
		}
		if pluginArray, ok := plugin.([]any); ok && len(pluginArray) >= 2 {
			if pluginStr, ok := pluginArray[0].(string); ok && pluginStr == "expo-build-properties" {
				hasBuildPropertiesPlugin = true

				// Check the plugin configuration
				if pluginConfig, ok := pluginArray[1].(map[string]any); ok {
					// Check iOS useFrameworks
					if iosConfig, exists := pluginConfig["ios"]; exists {
						if iosMap, ok := iosConfig.(map[string]any); ok {
							if useFrameworks, exists := iosMap["useFrameworks"]; exists {
								if frameworks, ok := useFrameworks.(string); ok && frameworks == "static" {
									hasIOSUseFrameworks = true
								}
							}
						}
					}

					// Check Android extraMavenRepos
					if androidConfig, exists := pluginConfig["android"]; exists {
						if androidMap, ok := androidConfig.(map[string]any); ok {
							if repos, exists := androidMap["extraMavenRepos"]; exists {
								notifeeRepo := "../../node_modules/@notifee/react-native/android/libs"
								if repoArray, ok := repos.([]any); ok {
									for _, repo := range repoArray {
										if repoStr, ok := repo.(string); ok && repoStr == notifeeRepo {
											hasAndroidExtraMavenRepos = true
											break
										}
									}
								} else if repoSlice, ok := repos.([]string); ok {
									for _, repo := range repoSlice {
										if repo == notifeeRepo {
											hasAndroidExtraMavenRepos = true
											break
										}
									}
								}
							}
						}
					}
				}
				break
			}
		}
	}

	if !hasBuildPropertiesPlugin {
		issues = append(issues, "expo-build-properties plugin not configured in app.json")
	} else {
		if !hasIOSUseFrameworks {
			issues = append(issues, "iOS useFrameworks not set to 'static' in expo-build-properties")
		}
		if !hasAndroidExtraMavenRepos {
			issues = append(issues, "Android extraMavenRepos missing Notifee path in expo-build-properties")
		}
	}

	// Check for Firebase configuration in android and ios sections
	if config.Expo.Android != nil {
		if _, exists := config.Expo.Android["googleServicesFile"]; !exists {
			issues = append(issues, "googleServicesFile not configured in app.json android section")
		}
		if _, exists := config.Expo.Android["package"]; !exists {
			issues = append(issues, "Android package name not configured in app.json")
		}
	} else {
		issues = append(issues, "Android configuration missing in app.json")
	}

	if config.Expo.IOS != nil {
		if _, exists := config.Expo.IOS["googleServicesFile"]; !exists {
			issues = append(issues, "googleServicesFile not configured in app.json ios section")
		}
		if _, exists := config.Expo.IOS["bundleIdentifier"]; !exists {
			issues = append(issues, "iOS bundle identifier not configured in app.json")
		}

		// Check for iOS push notification settings
		if entitlements, exists := config.Expo.IOS["entitlements"]; exists {
			if entMap, ok := entitlements.(map[string]any); ok {
				if _, hasAPS := entMap["aps-environment"]; !hasAPS {
					issues = append(issues, "aps-environment not configured in iOS entitlements")
				}
			}
		} else {
			issues = append(issues, "iOS entitlements not configured for push notifications")
		}

		if infoPlist, exists := config.Expo.IOS["infoPlist"]; exists {
			if plistMap, ok := infoPlist.(map[string]any); ok {
				if _, hasBackground := plistMap["UIBackgroundModes"]; !hasBackground {
					issues = append(issues, "UIBackgroundModes not configured in iOS infoPlist")
				}
			}
		} else {
			issues = append(issues, "iOS infoPlist not configured for background modes")
		}
	} else {
		issues = append(issues, "iOS configuration missing in app.json")
	}

	return issues
}

// CheckClixInitialization checks if Clix initialization file exists
func CheckClixInitialization(projectRoot string) bool {
	tsPath := filepath.Join(projectRoot, "clix-config.ts")
	jsPath := filepath.Join(projectRoot, "clix-config.js")

	_, tsErr := os.Stat(tsPath)
	_, jsErr := os.Stat(jsPath)

	return tsErr == nil || jsErr == nil
}

// CheckNativeCode checks if native code has been generated
func CheckNativeCode(projectRoot string) bool {
	androidPath := filepath.Join(projectRoot, "android")
	iosPath := filepath.Join(projectRoot, "ios")

	androidExists := false
	iosExists := false

	if info, err := os.Stat(androidPath); err == nil && info.IsDir() {
		androidExists = true
	}

	if info, err := os.Stat(iosPath); err == nil && info.IsDir() {
		iosExists = true
	}

	return androidExists && iosExists
}

// CheckClixAppIntegration checks if Clix is integrated into the App component
func CheckClixAppIntegration(projectRoot string) bool {
	// Common App component file paths in Expo projects
	appFiles := []string{
		"App.tsx",
		"App.js",
		"src/App.tsx",
		"src/App.js",
		"app/_layout.tsx", // Expo Router
		"app/_layout.js",  // Expo Router
		"src/app/_layout.tsx",
		"src/app/_layout.js",
	}

	for _, file := range appFiles {
		fullPath := filepath.Join(projectRoot, file)
		if content, err := os.ReadFile(fullPath); err == nil {
			appContent := string(content)
			// Check if Clix is imported and initialized
			hasImport := strings.Contains(appContent, "initializeClix")
			hasCall := strings.Contains(appContent, "initializeClix()")
			return hasImport && hasCall
		}
	}

	return false
}

// checkMMKVVersion validates if the correct MMKV version is installed for the React Native version
func checkMMKVVersion(projectRoot string, dependencies map[string]bool) bool {
	// Check if MMKV is installed at all
	if !dependencies["react-native-mmkv"] {
		return false
	}

	// Get package.json to check versions
	packageJSONPath := filepath.Join(projectRoot, "package.json")
	data, err := os.ReadFile(packageJSONPath)
	if err != nil {
		return false
	}

	var packageJSON map[string]any
	if err := json.Unmarshal(data, &packageJSON); err != nil {
		return false
	}

	// Get installed package versions
	var reactNativeVersion, mmkvVersion string
	if deps, ok := packageJSON["dependencies"].(map[string]any); ok {
		if rnVersion, exists := deps["react-native"]; exists {
			if versionStr, ok := rnVersion.(string); ok {
				reactNativeVersion = versionStr
			}
		}
		if mmkv, exists := deps["react-native-mmkv"]; exists {
			if versionStr, ok := mmkv.(string); ok {
				mmkvVersion = versionStr
			}
		}
	}

	if reactNativeVersion == "" || mmkvVersion == "" {
		return false
	}

	// Parse React Native version
	rnVersion, err := parseReactNativeVersionForDoctor(reactNativeVersion)
	if err != nil {
		return false
	}

	// Parse MMKV version to get major version
	mmkvMajor, err := parseMMKVMajorVersion(mmkvVersion)
	if err != nil {
		return false
	}

	// Check version compatibility
	switch {
	case rnVersion >= 74 && mmkvMajor >= 3:
		return true // RN 0.74+ should use MMKV 3.x
	case rnVersion < 74 && mmkvMajor == 2:
		return true // RN < 0.74 should use MMKV 2.x
	default:
		return false // Version mismatch
	}
}

// parseReactNativeVersionForDoctor parses React Native version for doctor checks
func parseReactNativeVersionForDoctor(versionStr string) (int, error) {
	// Remove common prefixes and suffixes
	version := strings.TrimPrefix(versionStr, "^")
	version = strings.TrimPrefix(version, "~")
	version = strings.TrimPrefix(version, ">=")
	version = strings.TrimPrefix(version, "<=")
	version = strings.TrimPrefix(version, ">")
	version = strings.TrimPrefix(version, "<")

	// Split by dots to get major.minor
	parts := strings.Split(version, ".")
	if len(parts) < 2 {
		return 0, fmt.Errorf("invalid version format: %s", versionStr)
	}

	major := strings.TrimSpace(parts[0])
	minor := strings.TrimSpace(parts[1])

	// Parse major version
	majorInt, err := strconv.Atoi(major)
	if err != nil {
		return 0, fmt.Errorf("invalid major version: %s", major)
	}

	// Parse minor version
	minorInt, err := strconv.Atoi(minor)
	if err != nil {
		return 0, fmt.Errorf("invalid minor version: %s", minor)
	}

	// Return as single integer (e.g., 0.74 -> 74, 0.75 -> 75)
	return majorInt*100 + minorInt, nil
}

// parseMMKVMajorVersion extracts the major version number from MMKV version string
func parseMMKVMajorVersion(versionStr string) (int, error) {
	// Remove common prefixes
	version := strings.TrimPrefix(versionStr, "^")
	version = strings.TrimPrefix(version, "~")
	version = strings.TrimPrefix(version, ">=")
	version = strings.TrimPrefix(version, "<=")
	version = strings.TrimPrefix(version, ">")
	version = strings.TrimPrefix(version, "<")

	// Split by dots to get major version
	parts := strings.Split(version, ".")
	if len(parts) < 1 {
		return 0, fmt.Errorf("invalid version format: %s", versionStr)
	}

	major := strings.TrimSpace(parts[0])

	// Parse major version
	majorInt, err := strconv.Atoi(major)
	if err != nil {
		return 0, fmt.Errorf("invalid major version: %s", major)
	}

	return majorInt, nil
}

File: src/packages/expo/doctor.ts
import { promises as fs } from 'fs';
import path from 'path';

// AppConfig represents the app.json configuration
interface AppConfig {
  expo: ExpoConfig;
}

interface ExpoConfig {
  name: string;
  slug: string;
  version: string;
  orientation: string;
  icon: string;
  userInterfaceStyle: string;
  splash: Record<string, any>;
  updates: Record<string, any>;
  assetBundlePatterns: string[];
  ios?: Record<string, any>;
  android?: Record<string, any>;
  web: Record<string, any>;
  plugins: any[];
}

export async function runExpoDoctor(): Promise<void> {
  const projectRoot = process.cwd();

  console.log('🔍 Running Clix Doctor for React Native Expo...');
  console.log('==================================================\n');

  const issues: string[] = [];

  // Check 1: Expo project structure
  if (!(await checkExpoProject(projectRoot))) {
    issues.push('Not an Expo project - missing app.json or expo dependency');
  } else {
    console.log('✅ Expo project detected');
  }

  // Check 2: Required dependencies
  const missingDeps = await checkDependencies(projectRoot);
  if (missingDeps.length > 0) {
    issues.push(`Missing dependencies: ${missingDeps.join(', ')}`);
  } else {
    console.log('✅ All required dependencies installed');
  }

  // Check 3: Firebase configuration files
  const hasAndroidConfig = await checkFirebaseConfig(projectRoot, 'android');
  const hasIOSConfig = await checkFirebaseConfig(projectRoot, 'ios');

  if (!hasAndroidConfig) {
    issues.push('Missing google-services.json file');
  } else {
    console.log('✅ google-services.json found');
  }

  if (!hasIOSConfig) {
    issues.push('Missing GoogleService-Info.plist file');
  } else {
    console.log('✅ GoogleService-Info.plist found');
  }

  // Check 4: app.json configuration
  const configIssues = await checkAppConfig(projectRoot);
  if (configIssues.length > 0) {
    issues.push(...configIssues);
  } else {
    console.log('✅ app.json properly configured');
  }

  // Check 5: Clix initialization file
  if (!(await checkClixInitialization(projectRoot))) {
    issues.push('Clix initialization file not found');
  } else {
    console.log('✅ Clix initialization file found');
  }

  // Check 6: Clix integration in App component
  if (!(await checkClixAppIntegration(projectRoot))) {
    issues.push('Clix not integrated in App component');
  } else {
    console.log('✅ Clix integrated in App component');
  }

  // Check 7: Generated native code
  if (!(await checkNativeCode(projectRoot))) {
    issues.push("Native code not generated - run 'npx expo prebuild --clean'");
  } else {
    console.log('✅ Native code generated');
  }

  // Report results
  console.log('');
  if (issues.length === 0) {
    console.log('🎉 All checks passed! Your Expo project is ready for Clix SDK.');
    console.log("You can now run 'npx expo run:android' or 'npx expo run:ios' to test push notifications.");
  } else {
    console.log(`❌ Found ${issues.length} issue(s):`);
    issues.forEach((issue, i) => {
      console.log(`  ${i + 1}. ${issue}`);
    });
    console.log('');
    console.log("Please fix the above issues and run 'clix doctor --expo' again.");
  }
}

async function checkExpoProject(projectRoot: string): Promise<boolean> {
  const appJSONPath = path.join(projectRoot, 'app.json');
  try {
    await fs.stat(appJSONPath);
  } catch {
    return false;
  }

  // Check if package.json exists and contains expo
  const packageJSONPath = path.join(projectRoot, 'package.json');
  try {
    const data = await fs.readFile(packageJSONPath, 'utf-8');
    return data.includes('expo');
  } catch {
    return false;
  }
}

async function checkFirebaseConfig(projectRoot: string, platform: string): Promise<boolean> {
  let fileName: string;
  switch (platform) {
    case 'android':
      fileName = 'google-services.json';
      break;
    case 'ios':
      fileName = 'GoogleService-Info.plist';
      break;
    default:
      return false;
  }

  const configPath = path.join(projectRoot, fileName);
  try {
    await fs.stat(configPath);
    return true;
  } catch {
    return false;
  }
}

async function checkDependencies(projectRoot: string): Promise<string[]> {
  const requiredDeps = [
    'expo-dev-client',
    '@react-native-firebase/app',
    '@react-native-firebase/messaging',
    'expo-build-properties',
    '@clix-so/react-native-sdk',
    '@notifee/react-native',
    'react-native-device-info',
    'react-native-get-random-values',
    'uuid',
  ];

  const packageJSONPath = path.join(projectRoot, 'package.json');
  try {
    const data = await fs.readFile(packageJSONPath, 'utf-8');
    const packageJSON = JSON.parse(data);

    const dependencies: Record<string, boolean> = {};
    if (packageJSON.dependencies) {
      Object.keys(packageJSON.dependencies).forEach((dep) => {
        dependencies[dep] = true;
      });
    }
    if (packageJSON.devDependencies) {
      Object.keys(packageJSON.devDependencies).forEach((dep) => {
        dependencies[dep] = true;
      });
    }

    const missing: string[] = [];
    for (const dep of requiredDeps) {
      if (!dependencies[dep]) {
        missing.push(dep);
      }
    }

    // Check MMKV separately as it needs version-specific validation
    if (!(await checkMMKVVersion(projectRoot, dependencies))) {
      missing.push('react-native-mmkv (incorrect version)');
    }

    return missing;
  } catch {
    return requiredDeps; // Return all as missing if can't read package.json
  }
}

async function checkAppConfig(projectRoot: string): Promise<string[]> {
  const issues: string[] = [];

  const appJSONPath = path.join(projectRoot, 'app.json');
  try {
    const data = await fs.readFile(appJSONPath, 'utf-8');
    const config: AppConfig = JSON.parse(data);

    // Check for Firebase App plugin
    let hasFirebaseAppPlugin = false;
    let hasFirebaseMessagingPlugin = false;
    for (const plugin of config.expo.plugins) {
      if (typeof plugin === 'string') {
        if (plugin === '@react-native-firebase/app') {
          hasFirebaseAppPlugin = true;
        }
        if (plugin === '@react-native-firebase/messaging') {
          hasFirebaseMessagingPlugin = true;
        }
      }
      if (Array.isArray(plugin) && plugin.length > 0) {
        if (plugin[0] === '@react-native-firebase/app') {
          hasFirebaseAppPlugin = true;
        }
        if (plugin[0] === '@react-native-firebase/messaging') {
          hasFirebaseMessagingPlugin = true;
        }
      }
    }

    if (!hasFirebaseAppPlugin) {
      issues.push('@react-native-firebase/app plugin not configured in app.json');
    }

    if (!hasFirebaseMessagingPlugin) {
      issues.push('@react-native-firebase/messaging plugin not configured in app.json');
    }

    // Check for expo-build-properties plugin and its configuration
    let hasBuildPropertiesPlugin = false;
    let hasIOSUseFrameworks = false;
    let hasAndroidExtraMavenRepos = false;

    for (const plugin of config.expo.plugins) {
      if (typeof plugin === 'string' && plugin === 'expo-build-properties') {
        hasBuildPropertiesPlugin = true;
        // String plugin format doesn't have configuration, so these are missing
      }
      if (Array.isArray(plugin) && plugin.length >= 2) {
        if (plugin[0] === 'expo-build-properties') {
          hasBuildPropertiesPlugin = true;

          // Check the plugin configuration
          const pluginConfig = plugin[1];
          if (pluginConfig && typeof pluginConfig === 'object') {
            // Check iOS useFrameworks
            if (pluginConfig.ios && typeof pluginConfig.ios === 'object') {
              if (pluginConfig.ios.useFrameworks === 'static') {
                hasIOSUseFrameworks = true;
              }
            }

            // Check Android extraMavenRepos
            if (pluginConfig.android && typeof pluginConfig.android === 'object') {
              const repos = pluginConfig.android.extraMavenRepos;
              const notifeeRepo = '../../node_modules/@notifee/react-native/android/libs';
              if (Array.isArray(repos)) {
                if (repos.includes(notifeeRepo)) {
                  hasAndroidExtraMavenRepos = true;
                }
              }
            }
          }
          break;
        }
      }
    }

    if (!hasBuildPropertiesPlugin) {
      issues.push('expo-build-properties plugin not configured in app.json');
    } else {
      if (!hasIOSUseFrameworks) {
        issues.push("iOS useFrameworks not set to 'static' in expo-build-properties");
      }
      if (!hasAndroidExtraMavenRepos) {
        issues.push('Android extraMavenRepos missing Notifee path in expo-build-properties');
      }
    }

    // Check for Firebase configuration in android and ios sections
    if (config.expo.android) {
      if (!config.expo.android.googleServicesFile) {
        issues.push('googleServicesFile not configured in app.json android section');
      }
      if (!config.expo.android.package) {
        issues.push('Android package name not configured in app.json');
      }
    } else {
      issues.push('Android configuration missing in app.json');
    }

    if (config.expo.ios) {
      if (!config.expo.ios.googleServicesFile) {
        issues.push('googleServicesFile not configured in app.json ios section');
      }
      if (!config.expo.ios.bundleIdentifier) {
        issues.push('iOS bundle identifier not configured in app.json');
      }

      // Check for iOS push notification settings
      if (config.expo.ios.entitlements && typeof config.expo.ios.entitlements === 'object') {
        if (!config.expo.ios.entitlements['aps-environment']) {
          issues.push('aps-environment not configured in iOS entitlements');
        }
      } else {
        issues.push('iOS entitlements not configured for push notifications');
      }

      if (config.expo.ios.infoPlist && typeof config.expo.ios.infoPlist === 'object') {
        if (!config.expo.ios.infoPlist.UIBackgroundModes) {
          issues.push('UIBackgroundModes not configured in iOS infoPlist');
        }
      } else {
        issues.push('iOS infoPlist not configured for background modes');
      }
    } else {
      issues.push('iOS configuration missing in app.json');
    }

    return issues;
  } catch {
    issues.push('Could not read or parse app.json');
    return issues;
  }
}

async function checkClixInitialization(projectRoot: string): Promise<boolean> {
  const tsPath = path.join(projectRoot, 'clix-config.ts');
  const jsPath = path.join(projectRoot, 'clix-config.js');

  try {
    await fs.stat(tsPath);
    return true;
  } catch {
    try {
      await fs.stat(jsPath);
      return true;
    } catch {
      return false;
    }
  }
}

async function checkNativeCode(projectRoot: string): Promise<boolean> {
  const androidPath = path.join(projectRoot, 'android');
  const iosPath = path.join(projectRoot, 'ios');

  let androidExists = false;
  let iosExists = false;

  try {
    const info = await fs.stat(androidPath);
    if (info.isDirectory()) {
      androidExists = true;
    }
  } catch {
    androidExists = false;
  }

  try {
    const info = await fs.stat(iosPath);
    if (info.isDirectory()) {
      iosExists = true;
    }
  } catch {
    iosExists = false;
  }

  return androidExists && iosExists;
}

async function checkClixAppIntegration(projectRoot: string): Promise<boolean> {
  // Common App component file paths in Expo projects
  const appFiles = [
    'App.tsx',
    'App.js',
    'src/App.tsx',
    'src/App.js',
    'app/_layout.tsx', // Expo Router
    'app/_layout.js', // Expo Router
    'src/app/_layout.tsx',
    'src/app/_layout.js',
  ];

  for (const file of appFiles) {
    const fullPath = path.join(projectRoot, file);
    try {
      const content = await fs.readFile(fullPath, 'utf-8');
      // Check if Clix is imported and initialized
      const hasImport = content.includes('initializeClix');
      const hasCall = content.includes('initializeClix()');
      return hasImport && hasCall;
    } catch {
      continue;
    }
  }

  return false;
}

async function checkMMKVVersion(projectRoot: string, dependencies: Record<string, boolean>): Promise<boolean> {
  // Check if MMKV is installed at all
  if (!dependencies['react-native-mmkv']) {
    return false;
  }

  // Get package.json to check versions
  const packageJSONPath = path.join(projectRoot, 'package.json');
  try {
    const data = await fs.readFile(packageJSONPath, 'utf-8');
    const packageJSON = JSON.parse(data);

    // Get installed package versions
    let reactNativeVersion = '';
    let mmkvVersion = '';
    if (packageJSON.dependencies) {
      if (packageJSON.dependencies['react-native']) {
        reactNativeVersion = packageJSON.dependencies['react-native'];
      }
      if (packageJSON.dependencies['react-native-mmkv']) {
        mmkvVersion = packageJSON.dependencies['react-native-mmkv'];
      }
    }

    if (!reactNativeVersion || !mmkvVersion) {
      return false;
    }

    // Parse React Native version
    const rnVersion = parseReactNativeVersionForDoctor(reactNativeVersion);
    if (rnVersion === null) {
      return false;
    }

    // Parse MMKV version to get major version
    const mmkvMajor = parseMMKVMajorVersion(mmkvVersion);
    if (mmkvMajor === null) {
      return false;
    }

    // Check version compatibility
    if (rnVersion >= 74 && mmkvMajor >= 3) {
      return true; // RN 0.74+ should use MMKV 3.x
    } else if (rnVersion < 74 && mmkvMajor === 2) {
      return true; // RN < 0.74 should use MMKV 2.x
    } else {
      return false; // Version mismatch
    }
  } catch {
    return false;
  }
}

function parseReactNativeVersionForDoctor(versionStr: string): number | null {
  // Remove common prefixes and suffixes
  let version = versionStr
    .replace(/^\^/, '')
    .replace(/^~/, '')
    .replace(/^>=/, '')
    .replace(/^<=/, '')
    .replace(/^>/, '')
    .replace(/^</, '');

  // Split by dots to get major.minor
  const parts = version.split('.');
  if (parts.length < 2) {
    return null;
  }

  const major = parts[0].trim();
  const minor = parts[1].trim();

  // Parse major version
  const majorInt = parseInt(major, 10);
  if (isNaN(majorInt)) {
    return null;
  }

  // Parse minor version
  const minorInt = parseInt(minor, 10);
  if (isNaN(minorInt)) {
    return null;
  }

  // Return as single integer (e.g., 0.74 -> 74, 0.75 -> 75)
  return majorInt * 100 + minorInt;
}

function parseMMKVMajorVersion(versionStr: string): number | null {
  // Remove common prefixes
  let version = versionStr
    .replace(/^\^/, '')
    .replace(/^~/, '')
    .replace(/^>=/, '')
    .replace(/^<=/, '')
    .replace(/^>/, '')
    .replace(/^</, '');

  // Split by dots to get major version
  const parts = version.split('.');
  if (parts.length < 1) {
    return null;
  }

  const major = parts[0].trim();

  // Parse major version
  const majorInt = parseInt(major, 10);
  if (isNaN(majorInt)) {
    return null;
  }

  return majorInt;
}
